src/aap_eda/core/models/activation.py

```diff
@@ -74,6 +80,46 @@ class Meta:
     status_updated_at = models.DateTimeField(null=True)
     status_message = models.TextField(null=True, default=None)
 
+    def save(self, *args, **kwargs):
+        # when creating
+        if self._state.adding:
+            if self.status_message is None:
+                self._set_status_message()
+        else:
+            if kwargs is None:
+                raise ValidationError(
+                    _(
+                        "update_fields is required to use when saving "
+                        "due to race conditions"
+                    )
+                )
+            # when updating without status
+            elif (
+                "status_message" in kwargs["update_fields"]
+                and "status" not in kwargs["update_fields"]
+            ):
+                raise ValidationError(
```

Comment:
This is not a validation error. Can you use a custom exception StatusMessageRequiredError?